client/src/app/my-projects/url-store.ts

```diff
@@ -0,0 +1,20 @@
+import { ACTIVITY } from "@shared/entities/activity.enum";
+import { parseAsJson, useQueryState } from "nuqs";
+import { z } from "zod";
+
+export const filtersSchema = z.object({
+  activity: z.array(z.nativeEnum(ACTIVITY)),
+  keyword: z.string().optional(),
+});
+
+export const INITIAL_FILTERS_STATE: z.infer<typeof filtersSchema> = {
+  activity: [],
+  keyword: "",
+};
+
+export function useMyProjectsFilters() {
+  return useQueryState(
+    "filters",
+    parseAsJson(filtersSchema.parse).withDefault(INITIAL_FILTERS_STATE),
+  );
+}
```

Comment:
Let's keep it but I don't see any value of keeping the state of the page in the URL in a page that only the user can access. The page itself is only accessible by the user, so making it shareable could have been optional in this case.

Reply:
A user can bookmark his filters in the browser, other than that I agree with you